server/users/routes.js

```diff
@@ -4,11 +4,12 @@ import { getUsers, addUser } from './controller';
 export const userRouter = express.Router();
 
 userRouter.get('/', async(req, res) => {
-  await getUsers();
+  await getUsers(req, res);
   res.status(200).send('get users');
 });
 
-userRouter.get('/add', async(req, res) => {
-  await addUser();
-  res.status(200).send('add users');
+userRouter.post('/add', async(req, res) => {
+  await addUser(req, res);
+  // res.status(200).send('add users');
+  // commented this out. Browser complains that we can't set headers twice.
```

Comment:
Ideally we separate out our request logic from our controller logic - we can get the parameters/bodies here

This has the benefit of making our controllers more independent and more testable - they don't need access to the HTTP context and can work with internal objects, and they can return objects/errors that we can handle at the router level

Reply:
Ok that makes a lot of sense.  I did the separation.  I tested the POST and GET functions and they both work. I'm not sure if I separated them how you explained. We still need to add error handling.

Reply:
Ok I think I have a better understanding of this now. Here's the user route. 
![image](https://user-images.githubusercontent.com/35232697/105070881-87a85300-5a49-11eb-9e4a-fc068760faca.png)